getzig

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+#
+# Download the binary archive of the latest Zig toolchain and install it
+# if it is not already installed.
+#-----------------------------------------------------------------------------
+#
+# Where the Zig toolchain is installed
+BASE_DIR=/opt/zig
+
+: "${GETZIG_DEBUG:=0}"
+: "${GETZIG_DRYRUN:=0}"
+
+case $(uname) in
+Linux) OS=linux ;;
+Darwin) OS=macos ;;
+*)
+	printf 'Unknown OS name: %s\n' "$(uname)" >&2
+	exit 1
+	;;
+esac
+
+case $(uname -m) in
+x86_64) ARCH=x86_64 ;;
+aarch64) ARCH=aarch64 ;;
+arm64) ARCH=aarch64 ;;
+arm*) ARCH=armv7a ;;
+*)
+	printf 'Unknown architecture: %s\n' "$(uname -m)" >&2
+	exit 1
+	;;
+esac
+
+#-----------------------------------------------------------------------------
+main() {
+	set -euo pipefail
+	declare -A info
+	local release="${1-stable}"
+
+	while IFS='=' read -r key value; do
+		info["${key}"]="${value}"
+	done <<<"$(get_latest_info "${release}")"
+
+	version="${info[version]}"
+	install_dir="${BASE_DIR}/${version}"
+	if [[ -e "${install_dir}" ]]; then
+		printf '%s version %s is up to date\n' "${release}" "${version}"
+		exit 0
+	fi
+
+	printf 'Downloading latest %s version %s\n' "${release}" "${version}"
+	mkdir "${install_dir}"
+	run curl -sL -o - "${info[tarball]}" | run tar -C "${install_dir}" -x -J --strip-components=1
+	run mkdir "${install_dir}/bin"
+	run mv "${install_dir}/zig" "${install_dir}/bin"
+	run ln -nsf "${version}" "${BASE_DIR}/${release}"
+}
+
+#-----------------------------------------------------------------------------
+get_latest_info() {
+	local select_release=''
+	local select_file=".[\"${ARCH}-${OS}\"] + {version: .version}"
+	local to_key_val='to_entries | .[] | .key + "=" + (.value | tostring)'
+
+	case "${1-}" in
+	master)
+		select_release='.master'
+		;;
+	stable | '')
+		# Note: this relies on jq processing map entries in the order they appear. This
+		# may not be the case with gojq
+		select_release='to_entries | .[1] | {version: .key} + .value'
+		;;
+	*)
+		printf 'unknown release type. Must be stable (default) or master\n' >&2
+		exit 1
+		;;
+	esac
+
+	wetrun curl -sL "https://ziglang.org/download/index.json" |
```

Comment:
🍭

Reply:
to me wetrun is a confusing term. alwaysrun or forcerun would make more sense to me.
(you choose dry, so i would have expected that you chose wet)...
feel free to ignore if this is a common pattern / doesn't resonate.

Reply:
ok. i changed it to `alwaysrun`